### DISCUSSION TITLE
How to use RegexManager for package.json?

### DISCUSSION THREAD

**Discussion by chkpnt** [renovatebot/renovate#20703] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.97.6

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have the following in my `package.json`:

```json5
{
    // ...
    "dependencies": {
        // ...
        "core-js": "3.28.0",
        // ...
    },
    // ...
    "babel": {
        "presets": [
            [
                "@babel/preset-env",
                {
                    "useBuiltIns": "usage",
                    "corejs": {
                        "version": "3.28.0",
                        "proposals": false
                    },
                    "debug": false
                }
            ]
        ]
    },
    // ...
}
```

I'd like to update the corejs-version within the babel-configuration together with the `core-js` dependency. Unfortunately, as the `package.json` doesn't support comments, I can't add something like `// renovate: datasource=npm depName=core-js` to the corresponding line.

What's the best practice on how to use the RegexManager in such a case? Sure, I could hard-code the `matchStrings` for the specific use-case. But is there something else?

### What you tried so far.

-

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
In this case you need to have a single regex manager per dependency like this, and you can use static strings for datasourceTemplate, depNameTemplate, etc.

**Reply by chkpnt** [reply]
Thanks for the confirmation.

For my specific issue with `core-js`, this is my solution:

```json5
    // ...
    packageRules: [
        {
            matchManagers: ["npm", "regex"],
            matchDepTypes: ["dependencies"],
            matchUpdateTypes: ["minor", "patch"],
            groupName: "dependencies (GUI)",
        },
        {
            matchManagers: ["npm"],
            matchDepTypes: ["devDependencies"],
            matchUpdateTypes: ["minor", "patch"],
            groupName: "devDependencies (GUI)",
        },
    ],
    regexManagers: [
        {
            fileMatch: ["(^|/)package.json$"],
            matchStringsStrategy: "recursive",
            matchStrings: [
                '"babel":\\s*{[^}]*}',
                '"corejs":\\s*\\{[^}]*}',
                '"version":\\s*"(?<currentValue>.*)"',
            ],
            depNameTemplate: "core-js",
            depTypeTemplate: "dependencies",
            datasourceTemplate: "npm",
        },
    ],
    // ...
```
